### ISSUE TITLE
the pictures I trained are not very good. I don't know what the problem is.can anyone help me？ thank you！！

### ISSUE THREAD

**Issue by FFjiahao** [costapt/adversarial_retinal_synthesis#13]
the pictures I trained are not very good. I don't know what the problem is.
![image](https://user-images.githubusercontent.com/62247114/158537940-deaf2c2e-3539-4f35-9e46-29abb77c4c27.png)

![image](https://user-images.githubusercontent.com/62247114/158537846-64d0d7a3-6353-4440-8d6d-f855a3aa462e.png)
![image](https://user-images.githubusercontent.com/62247114/158538026-739bbe36-8496-4ff9-9931-a71a6f313eb1.png)
![image](https://user-images.githubusercontent.com/62247114/158538066-93d0e202-65f8-4c09-ac96-f69df93f6de0.png)
![image](https://user-images.githubusercontent.com/62247114/158538103-7e88cfbf-9e57-4bd5-952f-22ce8e12c0d5.png)

![image](https://user-images.githubusercontent.com/62247114/158538130-a0bb6b77-ea55-4610-a448-59c3e74a28e2.png)
![image](https://user-images.githubusercontent.com/62247114/158538157-283a63bc-6230-4bb0-8b6e-ce4615664d16.png)

**Comment by FFjiahao** [comment]
@costapt @diazandr3s  Whether it's related to the data set？Looking forward to your reply.thank you.

**Comment by AliSaeed86** [comment]
Hi @FFjiahao,
Which dataset have you used please?
i ran the code using DRIVE dataset with 16 images for training and 4 images for validation.

Regards

**Comment by FFjiahao** [comment]
@AliSaeed86 .I use messidor  dataset.How does your code work?Can you send your training results?

**Comment by AliSaeed86** [comment]
@FFjiahao
Actually i am not sure about the performance of my code. i just wanted to make it run then i thought to change the dataset with larger size one.
using the 16 images for training and 4 images for validation i got this atob.png image as this :
![atob](https://user-images.githubusercontent.com/68149304/158572205-b8bcf1e3-a0a0-4620-a843-b92caec40472.png)

and the result.png as this 
![results](https://user-images.githubusercontent.com/68149304/158572297-9a8193d1-a6f2-4f3a-8522-998aa6109721.png)

it seems that it is memorizing the dataset and generating same image for all, and the vessel tree is not generated.

i guess this issue is because i have used very few number of images for training and validating and not because of the code problem.

Please, how did you get the vessel masks of Messidor dataset?
have you excluded those images of grades 3 and 4 of diabetic retinopathy? and how did you recognize them from grades 0,1 and 2 ?

Thanks

**Comment by FFjiahao** [comment]
I've had the same results a few times before.This problem is not caused by too few pictures.See if you're using tif for the image suffix.You can use UNET network to segment blood vessels in datasets.i dont exclude the images of grades 3 and 4 of diabetic retinopathy.There are corresponding documents to mark the pictures of retinopathy，you can use the documents to classify pictures.

**Comment by FFjiahao** [comment]
@AliSaeed86 I've had the same results a few times before.This problem is not caused by too few pictures.See if you're using tif for the image suffix.You can use UNET network to segment blood vessels in datasets.i dont exclude the images of grades 3 and 4 of diabetic retinopathy.There are corresponding documents to mark the pictures of retinopathy，you can use the documents to classify pictures.

**Comment by FFjiahao** [comment]
@AliSaeed86  can you tell me your version of keras,tensorflow and python? thank you.

**Comment by FFjiahao** [comment]
@AliSaeed86  can you tell me your version of keras,tensorflow and python? thank you.

**Comment by AliSaeed86** [comment]
@FFjiahao
yes i am using .tif images, do i have to change the extension to another type of images?
As i know, images in Messidor dataset aslo have .tif  extension.

i am using keras keras 1.2.2 , tensorflow 1.15 and python 3.7.11
i couldn't modify the code to work on tensorflow 2.x so that i built new envo with same configurations as Costa used.
if you could modify the code to earlier version of tensorflow and keras, please let me know.

Thanks

**Comment by AliSaeed86** [comment]
if you don't mind, please send me you email on ali.1986.kasim@gmail.com for further discussion. 
Thanks

**Comment by yichuan1998** [comment]
> [@AliSaeed86](https://github.com/AliSaeed86) can you tell me your version of keras,tensorflow and python? thank you.

Hey, could you share your code with me, I can't run the code with tf 1.15.0、keras1.2.2
